src/core/SamplePlayHandle.cpp

```diff
@@ -49,34 +49,20 @@ SamplePlayHandle::SamplePlayHandle( const QString& sampleFile ) :
 
 
 
-SamplePlayHandle::SamplePlayHandle( SampleBuffer* sampleBuffer ) :
-	PlayHandle( TypeSamplePlayHandle ),
-	m_sampleBuffer( sharedObject::ref( sampleBuffer ) ),
-	m_doneMayReturnTrue( true ),
-	m_frame( 0 ),
-	m_ownAudioPort( true ),
-	m_defaultVolumeModel( DefaultVolume, MinVolume, MaxVolume, 1 ),
-	m_volumeModel( &m_defaultVolumeModel ),
-	m_track( NULL ),
-	m_bbTrack( NULL )
+SamplePlayHandle::SamplePlayHandle( const QString& sampleFile ) :
+	SamplePlayHandle( new SampleBuffer( sampleFile ) )
 {
+	sharedObject::unref( m_sampleBuffer );
 	setAudioPort( new AudioPort( "SamplePlayHandle", false ) );
 }
 
 
 
 
 SamplePlayHandle::SamplePlayHandle( SampleTCO* tco ) :
-	PlayHandle( TypeSamplePlayHandle ),
-	m_sampleBuffer( sharedObject::ref( tco->sampleBuffer() ) ),
-	m_doneMayReturnTrue( true ),
-	m_frame( 0 ),
-	m_ownAudioPort( false ),
```

Comment:
After this PR, `m_ownAudioPort` is always set to true and cause crash when (un)mute sample track while playing. #3947 will fix it.